### ISSUE TITLE
Best Practice For Redirecting With Dynamic Context?

### ISSUE THREAD

**Issue by oldmanfleming** [react-boilerplate/react-boilerplate#2920]
## Description
I have two pages, one that lists items and one that you can edit and view an individual item. When I click on an item in the list I would like to do a redirect to the individual item page. How do I pass context of which item was clicked when I do the redirect to the page that loads the individual item? I am trying to do this two different ways and not sure which is correct:

One would be to add the id of the item to the route that I push to the connected router and then extract that id from the route in the individual item page on mount. This works in practice but  adding and extracting route params seems complex.

The other would be to dispatch an action to put the item clicked in the individual pages reducer before dispatching the redirect to that page. But in practice this has weird behavior. It seems like dispatching actions to a page that hasn't been loaded yet doesn't work? (It works locally but not in the production version)

Is one of these correct approaches? Is there a third option I am missing?

Any insight would be awesome!

## Steps to reproduce

**Expected behavior**

**Screenshots**

## Versions

- React-Boilerplate: 4.0
- Node/NPM:
- Browser:

**Comment by oldmanfleming** [comment]
Update: I've managed to solve the issue by removing the async loading from the individual rule page. Is this a reasonable way to handle this problem?

**Comment by oldmanfleming** [comment]
Second update: I determined that my initial update of removing the async loading was not a real fix. It appears that when I have Redux Dev Tools enabled, all page reducers/sagas seem to be loaded at once, but with Redux Dev Tools off this is not the case. So trying to pass state between pages if I have not yet visited the page only works with Redux Dev Tools on, which is weird.

Anyways, I have decided to go with my second approach of passing context through route params when navigating between pages as this seems to be the only way I can think to reliably do it.

I'll close this issue.
